packages/sdk/app-framework/src/plugins/plugin-settings/SettingsPlugin.tsx

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { RootSettingsStore } from '@dxos/local-storage';
+
+import SettingsMeta from './meta';
+import { type SettingsPluginProvides } from './provides';
+import { type PluginDefinition } from '../plugin-host';
+
+export const SettingsPlugin = (): PluginDefinition<SettingsPluginProvides> => {
+  // Global settings singleton.
+  const settingsStore = new RootSettingsStore();
```

Comment:
// TODO: Make settings storage extensible so that other plugins can decide the persistence of them.

(just as a reminder to not pull that into this layer)